src/api/entitlements.js

```diff
@@ -133,21 +161,46 @@ export class Entitlements {
   /**
    * Returns the first matching entitlement for the specified product,
    * optionally also matching the specified source.
+   *
+   * Returns non-metering entitlements if possible, to avoid using
+   * metered reads unnecessarily.
+   *
    * @param {?string} product
    * @param {string=} source
    * @return {?Entitlement}
    */
   getEntitlementFor(product, source) {
-    if (product && this.entitlements.length > 0) {
-      for (let i = 0; i < this.entitlements.length; i++) {
-        if (
-          this.entitlements[i].enables(product) &&
-          (!source || source == this.entitlements[i].source)
-        ) {
-          return this.entitlements[i];
-        }
+    if (!product) {
+      return null;
+    }
+
+    // Sort non-metering entitlements first, so they are returned if possible,
+    // to avoid using metered reads unnecessarily.
```

Comment:
ditto. why do they need to be sorted? is there a thing that will go through them in a specific order? what is that thing?

Reply:
I rewrote this comment and refactored the code below so it's (hopefully) more descriptive